docs/source/language_specification/query_directives.rst

```diff
@@ -629,7 +629,127 @@ Constraints and Rules
 @output_source
 --------------
 
-The documentation for this directive is in development.
+:code:`@output_source` is a directive that can be used on the last vertex field in a query to
+reverse the order in which vertex fields are visited. Currently, its primary function is to help
+deal with the following known issue that occurs when compiling to :code:`gremlin`. When vertex
+fields are visited in a certain order in a GraphQL query, the compiler returns a `gremlin` that
+does not return the complete set of results promised by the query semantics. See the example use
+section for more details.
+
+Example use
+~~~~~~~~~~~
+
```

Comment:
Removed 
"To explain the completeness of returned results in more detail, assume
the database contains the following example graph:"